src/Features/Core/Portable/EditAndContinue/ActiveInstructionId.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EditAndContinue
+{
+    [DebuggerDisplay("{GetDebuggerDisplay(),nq}")]
+    internal readonly struct ActiveInstructionId : IEquatable<ActiveInstructionId>
+    {
+        public readonly Guid ModuleId;
+        public readonly int MethodToken;
+        public readonly int MethodVersion;
+        public readonly int ILOffset;
+
+        public ActiveInstructionId(Guid moduleId, int methodToken, int methodVersion, int ilOffset)
+        {
+            ModuleId = moduleId;
+            MethodToken = methodToken;
+            MethodVersion = methodVersion;
+            ILOffset = ilOffset;
+        }
+
+        public override bool Equals(object obj)
+            => obj is ActiveInstructionId && Equals((ActiveInstructionId)obj);
+
+        public bool Equals(ActiveInstructionId other)
+            => ILOffset == other.ILOffset &&
+               MethodToken == other.MethodToken &&
+               MethodVersion == other.MethodVersion &&
+               ModuleId.Equals(other.ModuleId);
```

Comment:
I can use ==. 

Yes, the ordering is just micro optimization. Since there will be usually a lot of active statements in the same module (project) while unlikely at the same IL offset and method.

---
In reply to: [162463296](https://github.com/dotnet/roslyn/pull/24316#discussion_r162463296) [](ancestors = 162463296)

Reply:
Added comment.

---
In reply to: [162748557](https://github.com/dotnet/roslyn/pull/24316#discussion_r162748557) [](ancestors = 162748557,162463296)

Reply:
Actually, let's just not clutter it with comments like that. I think it will be barely noticeable if it is reordered. It should be a general practice to write predicates in the most likely to fail first. I don't think we need to add comments on every boolean expression, unless it's really on a hot path and we measured it.

---
In reply to: [162748576](https://github.com/dotnet/roslyn/pull/24316#discussion_r162748576) [](ancestors = 162748576,162748557,162463296)